engine/jobmaster/dm/openapi.go

```diff
@@ -0,0 +1,251 @@
+
+	router.GET("/binlog/tasks/:task-name", wrapper.DMAPIGetBinlogOperator)
+
+	router.POST("/binlog/tasks/:task-name", wrapper.DMAPISetBinlogOperator)
+
+	router.GET("/config", wrapper.DMAPIGetJobConfig)
+
+	router.PUT("/config", wrapper.DMAPIUpdateJobConfig)
+
+	router.GET("/schema/tasks/:task-name", wrapper.DMAPIGetSchema)
+
+	router.PUT("/schema/tasks/:task-name", wrapper.DMAPISetSchema)
+
+	router.GET("/status", wrapper.DMAPIGetJobStatus)
+
+	router.PUT("/status", wrapper.DMAPIOperateJob)
+}
+
+// DMAPIGetJobStatus implements the api of get job status.
+func (jm *JobMaster) DMAPIGetJobStatus(c *gin.Context, params openapi.DMAPIGetJobStatusParams) {
+	var tasks []string
+	if params.Tasks != nil {
+		tasks = *params.Tasks
+	}
+	resp, err := jm.QueryJobStatus(c.Request.Context(), tasks)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	c.IndentedJSON(http.StatusOK, resp)
+}
+
+// DMAPIGetJobConfig implements the api of get job config.
+func (jm *JobMaster) DMAPIGetJobConfig(c *gin.Context) {
+	cfg, err := jm.GetJobCfg(c.Request.Context())
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	bs, err := cfg.Yaml()
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	c.IndentedJSON(http.StatusOK, string(bs))
+}
+
+// DMAPIUpdateJobConfig implements the api of update job config.
+func (jm *JobMaster) DMAPIUpdateJobConfig(c *gin.Context) {
+	// TODO: support update job config
+}
+
+// DMAPIOperateJob implements the api of operate job.
+func (jm *JobMaster) DMAPIOperateJob(c *gin.Context) {
+	var req openapi.OperateJobRequest
+	if err := c.Bind(&req); err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	var op dmpkg.OperateType
+	switch req.Op {
+	case openapi.OperateJobRequestOpPause:
+		op = dmpkg.Pause
+	case openapi.OperateJobRequestOpResume:
+		op = dmpkg.Resume
+	default:
+		_ = c.Error(errors.Errorf("unsupport op type '%s' for operate task", req.Op))
+		return
+	}
+
+	var tasks []string
+	if req.Tasks != nil {
+		tasks = *req.Tasks
+	}
+	err := jm.OperateTask(c.Request.Context(), op, nil, tasks)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	c.Status(http.StatusOK)
+}
+
+// DMAPIGetBinlogOperator implements the api of get binlog operator.
+func (jm *JobMaster) DMAPIGetBinlogOperator(c *gin.Context, taskName string, params openapi.DMAPIGetBinlogOperatorParams) {
+	req := &dmpkg.BinlogRequest{
+		Op:      pb.ErrorOp_List,
+		Sources: []string{taskName},
+	}
+	if params.BinlogPos != nil {
+		req.BinlogPos = *params.BinlogPos
+	}
+	resp, err := jm.Binlog(c.Request.Context(), req)
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	c.IndentedJSON(http.StatusOK, resp)
+}
+
+// DMAPISetBinlogOperator implements the api of set binlog operator.
+func (jm *JobMaster) DMAPISetBinlogOperator(c *gin.Context, taskName string) {
+	var req openapi.SetBinlogOperatorRequest
+	if err := c.Bind(&req); err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	r := &dmpkg.BinlogRequest{
+		Sources: []string{taskName},
```

Comment:
seems the field should be `Task`, but in e2e_dm_test.go we use `[]string{source1, source2}` as tasks. Very confusing

Reply:
In engine we use tasks, in DM we use source...